shap/explainers/_tree.py

```diff
@@ -1517,7 +1517,11 @@ def get_trees(self, data=None, data_missing=None):
                     self.children_default[i,j] = self.node_cright[i][j]
                 self.features[i,j] = self.node_sindex[i][j] & ((np.uint32(1) << np.uint32(31)) - np.uint32(1))
                 if self.node_cleft[i][j] >= 0:
-                    self.thresholds[i,j] = self.node_info[i][j]
+                    # Xgboost uses < for thresholds where shap uses <=
+                    # Move the threshold down by the smallest possible increment
+                    self.thresholds[i, j] = np.nextafter(self.node_info[i][j],
```

Comment:
Perhaps the real problem is that at https://github.com/slundberg/shap/blob/f88d4f28bc7a7a9e863d26bbf5362b1b83255c60/shap/explainers/_tree.py#L1023-L1024

`np.inf` should be `-np.inf` ?

Reply:
Yes, that should do it, will update that now.

Reply:
Adjusting the threshold at line 1024 creates a bug because this adjustment happens after node statistics get calculated (in Tree constructor), and it needs to happen before. I suggest removing the `less_than_or_equal` parameter (which  is currently only used for xgb) and adjusting thresholds as necessary in each constructor.

Also, for xgb the nudge has to happen in float32, nudging the threshold in float64 seems insufficient.

Reply:
Ah! I bet that is something I missed before. Good catch. I agree that sounds like a good approach.